### PR TITLE
Frontend improvements

### PR DESCRIPTION
**Pull request by jallier**
This PR will detail the improvements to the frontend to setup a framework from which we can more easily add stuff in the future.

This can be merged when:
- [x] Frontend has a complete basic structure
- [x] Using emotion or styled-components for css styling
- [x] There is a file containing the common reusable components
- [x] Login, Register and storing the JWT in a cookie work
- [x] There are private routes (routes you can only access with a valid token i.e everything except the login page)
- [x] Redux works correctly with typescript

### CODE REVIEW

**Comment by jallier** [comment]
We decided to add redux in the hopes that it will make adding more functionality to it later a little easier

**Comment by jallier** [comment]
I'll add the redux persistence in a new PR, this one was starting to get unruly

**Review by robert-calvert — APPROVED** [review]
This all looks good to me, awesome work! 

I did notice there's a few `console.log`s about, is this intentional?

File: frontend/package.json [hunk 1]
```diff
@@ -3,18 +3,23 @@
...
+    "@emotion/styled": "^10.0.6",
     "@types/jest": "23.3.10",
     "@types/node": "10.12.17",
     "@types/react": "^16.7.18",
     "@types/react-dom": "^16.0.11",
+    "@types/react-redux": "^7.0.1",
     "@types/react-router": "^4.4.3",
     "@types/react-router-dom": "^4.3.1",
-    "antd": "^3.12.1",
+    "antd": "^3.12.5",
```
Comment (jallier) [inline frontend/package.json]:
```suggestion
    "antd": "^3.13.0",
```

File: frontend/src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,61 @@
...
+import { Header } from './Components/Common/Header';
+import { PrivateRoute } from './Components/Common/PrivateRoute';
+import { createStore } from 'redux';
+import MainReducer from './Redux/authenticate/reducers';
+import { Provider, connect } from 'react-redux';
+import { AuthenticatedState } from './Redux/authenticate/types';
+
+const store = createStore(MainReducer);
+
+console.log('Store is: ', store.getState());
```
Comment (jallier) [inline frontend/src/App.tsx]: Remove

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -1,61 +1,61 @@
...
-      </Router>
-    );
-  }
+const publicRoute = () => {
+    return <div>This is a public route</div>;
+};
+
+class Main extends Component<{ authenticated: any }> {
+    render() {
+        console.log('Auth state is', this.props.authenticated);
```
Comment (jallier) [inline frontend/src/App.tsx]: Remove

**Pull request merged**